internal/controller/api/v1alpha1/host_types.go

```diff
@@ -17,25 +17,188 @@ limitations under the License.
+	MacAddress    string   `json:"macAddress,omitempty"`
+	Flags         []string `json:"flags"`
+	SpeedMbps     int64    `json:"speedMBPS,omitempty"`
+}
+
+type HostInstallationEligibility struct {
+	Eligible           bool     `json:"eligible,omitempty"`
+	NotEligibleReasons []string `json:"notEligibleReasons"`
+}
+
+type HostIOPerf struct {
+	// 99th percentile of fsync duration in milliseconds
+	SyncDurationMilliseconds int64 `json:"syncDurationMilliseconds,omitempty"`
+}
+
+type HostDisk struct {
+	DriveType               string                      `json:"driveType,omitempty"`
+	Vendor                  string                      `json:"vendor,omitempty"`
+	Name                    string                      `json:"name,omitempty"`
+	Path                    string                      `json:"path,omitempty"`
+	Hctl                    string                      `json:"hctl,omitempty"`
+	ByPath                  string                      `json:"byPath,omitempty"`
+	Model                   string                      `json:"model,omitempty"`
+	Wwn                     string                      `json:"wwn,omitempty"`
+	Serial                  string                      `json:"serial,omitempty"`
+	SizeBytes               int64                       `json:"sizeBytes,omitempty"`
+	Bootable                bool                        `json:"bootable,omitempty"`
+	Smart                   string                      `json:"smart,omitempty"`
+	InstallationEligibility HostInstallationEligibility `json:"installationEligibility,omitempty"`
+	IoPerf                  HostIOPerf                  `json:"ioPerf,omitempty"`
+}
+
+type HostBoot struct {
+	CurrentBootMode string `json:"currentBootMode,omitempty"`
+	PxeInterface    string `json:"pxeInterface,omitempty"`
+}
+
+type HostSystemVendor struct {
+	SerialNumber string `json:"serialNumber,omitempty"`
+	ProductName  string `json:"productName,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Virtual      bool   `json:"virtual,omitempty"`
+}
+
+type HostInventory struct {
+	// Name in REST API: timestamp
+	ReportTime   *metav1.Time     `json:"reportTime,omitempty"`
+	Hostname     string           `json:"hostname,omitempty"`
+	BmcAddress   string           `json:"bmcAddress,omitempty"`
+	BmcV6address string           `json:"bmcV6Address,omitempty"`
+	Memory       HostMemory       `json:"memory,omitempty"`
+	Cpu          HostCPU          `json:"cpu,omitempty"`
+	Interfaces   []HostInterface  `json:"interfaces"`
+	Disks        []HostDisk       `json:"disks"`
+	Boot         HostBoot         `json:"boot,omitempty"`
+	SystemVendor HostSystemVendor `json:"systemVendor,omitempty"`
+}
 
 // HostSpec defines the desired state of Host
 type HostSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// The name of the Cluster CR
+	ClusterRef              *corev1.ObjectReference `json:"clusterRef"`
```

Comment:
Devan pointed out on another PR that this field type has fallen out of favor, and the recommendation is to create your own reference type with whatever fields are meaningful to you. If we're assuming the same namespace, which I think is a reasonable starting point, this can just be a string field ending with "Name".

Also this is going to reference a hive ClusterDeployment per the latest agreed API, so maybe `ClusterDeploymentName`.

Reply:
@avishayt FYI. 
@mhrivnak is that the same for secret?

Reply:
added (and used) [here](https://github.com/openshift/assisted-service/pull/861/files#diff-068ee2662f5f4ed13f5a7a7f7e19dc2cfe822d8ed9877a8bb84217ea0d9eeac7R29).
currently, it has `Name` and `Namespace`

@filanov, let me know If I should place this struct elsewhere.